src/server.jl

```diff
@@ -327,39 +330,21 @@ function serve_file(
 end
 
 """
-    ws_upgrade(http::HTTP.Stream)
-
-Upgrade the HTTP request in the stream to a websocket.
-"""
-function ws_upgrade(http::HTTP.Stream)
-    # adapted from HTTP.WebSockets.upgrade; note that here the upgrade will always
-    # have  the right format as it always triggered by after a Response
-    HTTP.setstatus(http, 101)
-    HTTP.setheader(http, "Upgrade" => "websocket")
-    HTTP.setheader(http, "Connection" => "Upgrade")
-    key = HTTP.header(http, "Sec-WebSocket-Key")
-    HTTP.setheader(http, "Sec-WebSocket-Accept" => HTTP.WebSockets.hashedkey(key))
-    HTTP.startwrite(http)
-
-    return HTTP.WebSocket(http.stream, http.message, http.message.response; client=false)
-end
-
-
-"""
-    ws_tracker(ws::HTTP.WebSockets.WebSocket, target::AbstractString)
+    ws_tracker(ws::HTTP.WebSockets.WebSocket)
 
 Adds the websocket connection to the viewers in the global dictionary `WS_VIEWERS` to the entry
 corresponding to the targeted file.
 """
-function ws_tracker(ws::HTTP.WebSockets.WebSocket, target::AbstractString)
+function ws_tracker(ws::HTTP.WebSockets.WebSocket)
+    target = ws.request.target
```

Comment:
The `WebSocket` has both the request and response associated w/ the original `HTTP.Stream`, so we can access the `target` ourselves